Makefile

```diff
@@ -137,6 +137,11 @@ soak-eunit: couch
 	@$(REBAR) setup_eunit 2> /dev/null
 	while [ $$? -eq 0 ] ; do $(REBAR) -r eunit $(EUNIT_OPTS) ; done
 
+.PHONY: elixir
+elixir:
+	@rm -rf dev/lib
+	@dev/run -a adm:pass --no-eval test/elixir/run
```

Comment:
So it is impossible from an Elixir test to restart CouchDB? That is going to limit some of the kinds of tests we do in the JavaScript suite.

Why was this done, out of curiosity?

Reply:
I think we should be able to fix it. There are few options available. It is a matter of choosing the best one:
  - run script which reads commands from stdin (as it is done for JS tests)
  - similar thing in elixir which restarts beam and haproxy
  - simulation of a restart using `application:stop(couch)`/`application:start(couch)`
  - http endpoint `_admin/restart` or `_test/restart` (this would require some supervisor to deal with restart)

Elixir test suite is an integration test. When we run integration tests there is a need to do certain operations to facilitate black box testing. Namely SUT restart and failure injection. It would be nice to be able to reuse this test suite to test the installation (before we allow customers traffic) using integration test suite just to make sure everything is in order. Having said that I think that the best option would be to implement the following:
- add `--keep-runing` or `--with-restart` or `--supervise` option to `dev/run` script
- implement http endpoint (TBD) to control the beam from integration test suites.
  - we need to be able to disable this endpoint in production via either:
      - config entry
      - absence of environment variable
      - some other method which would make it hard to enable it by accident. 

This way we would have a similar restart behavior as in production. Since In production we run CouchDB from standard distribution specific supervisor which deals with restart.

Reply:
@iilyak I agree, I'd prefer not to use `application:stop/start` because that's not how CouchDB actually restarts in production most of the time. 

There used to be a restart endpoint, it was dropped with 2.x. The hack in `dev/run` is what @davisp put in to deal with its removal.  If you do put it back in, it should live under `/_node/<nodename>/_restart`.

Reply:
`POST /_node/<nodename>/_restart` is implemented here https://github.com/apache/couchdb/pull/1543